client/pages/Puzzle1Page/Puzzle1Page.jsx

```diff
@@ -20,17 +20,31 @@ export default function Puzzle1Page() {
   const [showOverlay, setShowOverlay] = useState(false);
   const [showTipsHint, setShowTipsHint] = useState(false);
   const [showComplete, setShowComplete] = useState(false);
+  const [timeOnSite, setTimeOnSite] = useState(0);
+  const [stoppedTime, setStoppedTime] = useState(0);
+  const [isPaused, setIsPaused] = useState(false);
   const [disableTipsButton, setDisableTipsButton] = useState(true);
   const [disableContactButton, setDisableContactButton] = useState(true);
   const handleCloseComplete = () => setShowComplete(false);
-  const handleShowComplete = () => setShowComplete(true);  
+  const intervalRef = useRef(null);
   const target = useRef(null);
+  const minutes = Math.floor(timeOnSite / 60);
+  const seconds = timeOnSite % 60;
+
+  // Keeps track of the time spent on the site, starts as soon as the page loads
+  // Time is paused once the puzzle is completed and the time is displayed in the completed modal
+  useEffect(() => {
+    const interval = setInterval(() => {
+      if(isPaused == false){
+        setTimeOnSite(timeOnSite => timeOnSite + 1);
+      }}, 1000);
+    return () => clearInterval(interval);
+  }, [isPaused]);
```

Comment:
Shouldn't this be a part of PuzzleNavBar for better reusability? I understand it would be a little cumbersome tossing around props, but we would reduce code duplication.

(Also, then you're not causing a rerender of _everything_ on the page when the time updates, only the PuzzleNavBar).